pkg/odo/genericclioptions/context.go

```diff
@@ -72,6 +72,42 @@ func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, er
 	}
 }
 
+func checkValidConfig(command *cobra.Command, configFileName string) {
+	_, err1 := config.NewLocalConfigInfo(configFileName, true)
+	allowedFirstChildCommandList := []string{"project", "catalog", "login", "logout"}
```

Comment:
maybe add a comment which explains that these commands dont need config and hence can run from directories which do not have a `.odo`

Reply:
Added